src/NServiceBus.Core/EndpointInfo.cs

```diff
@@ -0,0 +1,18 @@
+namespace NServiceBus
+{
+    using Settings;
+
+    class EndpointInfo
```

Comment:
So the values in this class are loaded and then never updated when/if they change in settings. Would there be a significant hit to perf if the property getters directly returned the appropriate `settings.Get...`? That would give you the nice wrapper still, but remove any potential caching problems.